docs/event-serializing.md

```diff
@@ -0,0 +1,139 @@
+---
+id: event-serializing
+title: Event Serializing
+sidebar_label: Event Serializing
+---
+
+We support [Apache Avro](https://avro.apache.org/) and standard JSON (de)serialization of events for producer as well as for consumer. This is working together with [Cloud Events specification](https://github.com/cloudevents/spec) (versions `0.1` and `0.2`). To make both of these worlds work we are utilizing Kafka headers introduced in version `0.11`. As Cloud Events specify there are 2 modes how to make this combination work. Discussion about this topic can be followed on [Github](https://github.com/cloudevents/spec/pull/337/files).
+
+**Structured content mode**
+
+This mode should be used when JSON format is desired, thus no further serialization and deserialization such as Apache Avro.
+
+> ##### KAFKA EVENT HEADERS
+>
+> ```
+> cloudEvent_contentType: "application/json"
+> ```
+>
+> ##### KAFKA EVENT VALUE
+>
+> ```
+> {
+>    "specversion" : "0.1",
+>    "type" : "com.example.someevent",
+>    "data": {},
+>    ...
+> }
+> ```
+
+**Binary content mode**
+
+This mode should be used when further serialization and deserialization is desired (e.g. Apache Avro).
+
+> ##### KAFKA EVENT HEADERS
+>
+> ```
+> cloudEvent_contentType: "avro/binary"
+> cloudEvent_specversion: "0.2"
+> cloudEvent_type: "com.example.someevent"
+> ...
+> ```
+>
+> ##### KAFKA EVENT VALUE
+>
+> ```
+> ... application data ...
+> ```
+
+## Apache Avro format
+
+To be compatible across the board Apache Avro is using specific [format](https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html#wire-format) that needs to be followed.
```

Comment:
isn't that like the main purpose of a format? :D

Reply:
I don't follow.

Reply:
well Avro is a format specification, so in order to be compatible with Avro - not surprisingly - you gotta follow the format spec. Just thought it's funny, no worries ;)